### PR TITLE
Extirpate sift

### PR DESCRIPTION
**Pull request by mnhrdt**
This moves the sift implementation from the base repository to github.com/cmla/sift
It also paves the way for moving out "homography" and most of the imscript files, simplifies the makefile and removes cmake and gsl dependencies.

### CODE REVIEW

**Comment by mnhrdt** [comment]
Note: the Dockerfile and the Manifest are still to be updated.  I do not know how to test for them.

File: .travis.yml [hunk 1]
```diff
@@ -9,22 +9,34 @@ matrix:
...
 before_install:
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-add-repository -y ppa:ubuntugis/ppa; fi
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get -qq update; fi
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install gdal-bin libfftw3-dev libgdal-dev geographiclib-tools libgeographic-dev libgsl-dev; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then HOMEBREW_NO_AUTO_UPDATE=1 brew install geographiclib fftw gsl; fi
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+      sudo apt-add-repository -y ppa:ubuntugis/ppa
+      sudo apt-get -qq update
+      sudo apt-get install gdal-bin libfftw3-dev libgdal-dev geographiclib-tools libgeographic-dev libgsl-dev
```
Comment (glostis) [inline .travis.yml]: Was `libgsl-dev` left here intentionally, or is it a mistake? I see it was removed from the README and from the macos dependencies a few lines below.

**Comment by glostis** [comment]
@mnhrdt I can take care of the Dockerfile and the MANIFEST.

File: .travis.yml (hunk 1 above)
Comment (mnhrdt) [reply .travis.yml]: it's a mistake, gsl is not necessary

Comment (carlodef) [reply .travis.yml]: I just removed it

**Comment by carlodef** [comment]
> @mnhrdt I can take care of the Dockerfile and the MANIFEST.

@glostis that would be great, thank you. Let's merge this pull request right now and address the Dockerfile and MANIFEST in a separate one.

**Pull request merged**